lib/pecorino/leaky_bucket.rb

```diff
@@ -90,12 +90,14 @@ def accepted?
   #   the bucket contents will then be capped at this value. So with
   #   bucket_capacity set to 12 and a `fillup(14)` the bucket will reach the level
   #   of 12, and will then immediately start leaking again.
-  def initialize(key:, capacity:, leak_rate: nil, over_time: nil)
+  # @param adapter[Pecorino::Adapters::BaseAdapter] a compatible adapter
+  def initialize(key:, capacity:, adapter: Pecorino.adapter, leak_rate: nil, over_time: nil)
```

Comment:
**question:** a question not related to this PR, more of a general question for LeakyBucket. I think our previous implementation allowed LeakyBucket to overfill, but you've fixed it, and now it doesn't do that any more.

If value is higher than capacity, it will refuse to fill a bucket.

I can imagine situations, where it would be great if leaky bucket could overfill. For example, if we're implementing a leaky bucket for API rate limiting, you might allow a user to exceed their rate limit by a small margin and then only enforce the limit if they continue to exceed it. This gives a bit of leeway for users who occasionally go over their limits but still protects your system from being overwhelmed by too many requests.

Reply:
A bucket can never overfill because all fillups get clamped to the capacity value. We have 2 methods - one for conditional fillup (where if your fillup "would" make the bucket overflow it gets refused) and one for unconditional fillup - it will only tell you whether the bucket did spillover as a result of your call. Having these two allows us to implement what you are describing, no? Or I am missing something and "Exceeding by a small margin" needs more information.

Reply:
> unconditional fillup - it will only tell you whether the bucket did spillover as a result of your call.

This is precisely what I was looking for.